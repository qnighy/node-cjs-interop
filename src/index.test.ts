import { describe, expect, it } from "@jest/globals";
import { transform } from "@babel/core";
import plugin from ".";

describe("babel-plugin-import-interop-only", () => {
  it("wraps default imports", () => {
    const result = transform(`import f from "mod";\nconsole.log(f);\n`, {
      presets: [],
      plugins: [plugin],
      configFile: false,
      babelrc: false,
    });
    expect(result?.code).toMatchSnapshot();
  });

  it("wraps named default imports", () => {
    const result = transform(`import { default as f } from "mod";\nconsole.log(f);\n`, {
      presets: [],
      plugins: [plugin],
      configFile: false,
      babelrc: false,
    });
    expect(result?.code).toMatchSnapshot();
  });

  it("wraps namespace imports", () => {
    const result = transform(`import * as M from "mod";\nconsole.log(M);\n`, {
      presets: [],
      plugins: [plugin],
      configFile: false,
      babelrc: false,
    });
    expect(result?.code).toMatchSnapshot();
  });
})
