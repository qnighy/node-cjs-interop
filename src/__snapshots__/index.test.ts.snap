// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-import-interop-only doesn't wrap named imports 1`] = `
"import { f } from \\"mod\\";
console.log(f);"
`;

exports[`babel-plugin-import-interop-only with typescript doesn't wrap named imports 1`] = `
"import { f } from \\"mod\\";
console.log(f);"
`;

exports[`babel-plugin-import-interop-only with typescript doesn't wrap type imports (default imports) 1`] = `"export {};"`;

exports[`babel-plugin-import-interop-only with typescript doesn't wrap type imports (named default imports) 1`] = `"export {};"`;

exports[`babel-plugin-import-interop-only with typescript doesn't wrap type imports (named default imports, individual specification) 1`] = `"export {};"`;

exports[`babel-plugin-import-interop-only with typescript doesn't wrap type imports (namespace imports) 1`] = `"export {};"`;

exports[`babel-plugin-import-interop-only with typescript wraps default imports 1`] = `
"const _f = _interopRequireDefault(f);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

import f from \\"mod\\";
console.log(_f.default);"
`;

exports[`babel-plugin-import-interop-only with typescript wraps named default imports 1`] = `
"const _f = _interopRequireDefault(f);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

import { default as f } from \\"mod\\";
console.log(_f.default);"
`;

exports[`babel-plugin-import-interop-only with typescript wraps namespace imports 1`] = `
"const _M = _interopRequireDefault(M);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

import * as M from \\"mod\\";
console.log(_M.default);"
`;

exports[`babel-plugin-import-interop-only wraps default imports 1`] = `
"const _f = _interopRequireDefault(f);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

import f from \\"mod\\";
console.log(_f.default);"
`;

exports[`babel-plugin-import-interop-only wraps named default imports 1`] = `
"const _f = _interopRequireDefault(f);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

import { default as f } from \\"mod\\";
console.log(_f.default);"
`;

exports[`babel-plugin-import-interop-only wraps namespace imports 1`] = `
"const _M = _interopRequireDefault(M);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

import * as M from \\"mod\\";
console.log(_M.default);"
`;
